src/externs.ts

```diff
@@ -79,26 +86,114 @@ export function getGeneratedExterns(externs: {[fileName: string]: string}): stri
   return allExterns;
 }
 
+/**
+ * Returns a mangled version of the module name (resolved file name) for source file.
+ *
+ * The mangled name is safe to use as a JavaScript identifier. It is used as a globally unique
+ * prefix to scope symbols in externs file (see code below).
+ *
+ * @param contextFileName if given is used as the context file path to resolve the source file's
+ *     name against.
+ */
+export function moduleNameAsIdentifier(
+    host: AnnotatorHost, fileName: string, contextFileName?: string): string {
+  const resolved = resolveModuleName(host, contextFileName || '', fileName);
+  return host.pathToModuleName('', resolved).replace(/\./g, '$');
```

Comment:
tsickle's basic assumption is that `pathToModuleName` generates unique, non-colliding module names. Otherwise our emit of `goog.module` and `goog.require` wouldn't work consistently.

I'm suspect (but haven't thought it through) that that assumption could be wrong with carefully crafted paths and module resolution configuration (trivially if you have different `baseUrl`s across files in a multi-step compilation, but could be more). Paths should always be unique in bazel though, where all paths are underneath the conceptual bazel root.